packages/frontend/src/components/ElectionForm/Races/RaceDialog.tsx

```diff
@@ -5,10 +5,13 @@ import { StyledButton } from '../../styles';
 import useElection from '../../ElectionContextProvider';
 
 
-export default function RaceDialog({ onSaveRace, open, handleClose, children }) {
+export default function RaceDialog({ onSaveRace, open, handleClose, children, editedRace }) {
+
 
     const { election } = useElection()
     const handleSave = () => {
+        //delete any blank candidates
+        editedRace.candidates = editedRace.candidates.filter(candidate => candidate.candidate_name !== '')
```

Comment:
nit

If there's a validation error onSave, will the blank candidate be added back?

Reply:
Do you want the blank candidate to be added back? I'm my testing it wasn't, but I can change it so it will